python/ray/train/_internal/data_config.py

```diff
@@ -120,6 +121,12 @@ def configure(
 
         return output
 
+    def _set_names(self, datasets: Dict[str, Dataset]):
+        """Set the name of each dataset if it is not already set."""
+        for name, ds in datasets.items():
+            if ds.name is None:
+                ds.set_name(name)
```

Comment:
Question: the train ds will be iterated over multiple times. And I think every epoch will create a new DS object. 

Will the name of each epoch be under the same name? or will it append some unique index?

Reply:
> Question: the train ds will be iterated over multiple times. And I think every epoch will create a new DS object.
Good point
> Will the name of each epoch be under the same name? or will it append some unique index?
there will be a UUID suffix for each run.